src/macros.rs

```diff
@@ -0,0 +1,9 @@
+#[macro_export]
```

Comment:
A small description comment would be nice.

Reply:
I will add it, I'm also not sure if I want to keep it actually, since now (after the refactoring) is used way less.